### DISCUSSION TITLE
How about extract all infomation needed by building parser to a json file?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [nvim-treesitter/nvim-treesitter#5588] [Ideas]
Currently, we have <https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lockfile.json>. However, some information in <https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/parsers.lua#L80-L2052> still be needed. Why not extract them to a universal json file? Such as:

```json
{
  "ada": {
    "revision": "f21f13afe0410311be1f78959a25aee393c569c9",
    "url": "https://github.com/briot/tree-sitter-ada",
    "files": [ "src/parser.c" ],
    "maintainers": [ "@briot" ]
  },
  ...
}
```

I think it can:

- Update <https://github.com/nvim-treesitter/nvim-treesitter#supported-languages> automatically by a perl/sed/jq script?
- Let other related projects share the list, e.g.,
  - <https://github.com/grantjenks/py-tree-sitter-languages/>
  - <https://github.com/foxfriends/syncat>
- Make easier to allow user to add support for new languages, even if they don't know lua, (only need they know json). It can easier <https://github.com/nvim-treesitter/nvim-treesitter/issues/2282>

**Comment by clason** [comment]
What would be the point? This information is needed for Lua functions to work, so why would we have this in a different format, just to force us to read and convert it on every startup?

> Update https://github.com/nvim-treesitter/nvim-treesitter#supported-languages automatically by a perl/sed/jq script?

That is already automatically updated using a Lua script.

> Make easier to allow user to add support for new languages, even if they don't know lua, (only need they know json). It can easier https://github.com/nvim-treesitter/nvim-treesitter/issues/2282

If you don't know Lua, you'll have a hard time configuring nvim-treesitter, even with only the bundled parsers.

**Reply by theHamsta** [reply]
@Freed-Wu yes, what you suggested was the plan some time as it would allow to reload the full configuration on the update of nvim-treesitter. But we never fully transitioned to a all JSON approach which causes breakages whenever a parser url is changed until the next Neovim restart. Probably we never completed this change because down sides are minor as @clason  mentioned and data other that revision is usually static.

Having the repo data in the JSON,
- Pro:
    -  fixes the reload logic
    - dependent data revision, repo, and parser files in one place (better data, code separation), when you add the parser you would automatically add the first revision
- Con:
    - mixing of human-authored and CI generated data  

There is a `main` branch with changes to the install logic. Please feel free to also suggest your discuss this topic in our Matrix channel which enables more real time discussion. I personally think both approaches would work well for most users and also maintainers of the repo.

**Reply by Freed-Wu** [reply]
Moreover, I even think tree-sitter should provide a repo to provide all information of all tree-sitter repo's url, revison, files, ... and update it on schedule.

**Reply by clason** [reply]
We do. In **this** repo. What would be the point for moving something that is the critical foundation for this plugin to a separate plugin?
